### PR TITLE
Multiline Business Input and Business Picker from existsting reports

### LINKED ISSUE #129: Populate list of businesses using existing boarding reports
image

The list of businesses in this view should be populated by searching the existing contact reports.

The search should be on the business name.

The business name + address should be used to identify the set of distinct businesses (e.g. if business "Oh My Cod" has franchises in different cities then they should be shown as distinct options here).

### PR DESCRIPTION
**Pull request by kojiba**
[-] Grouped ChooseBusiness in separate directory
[-] Added the states with loading/loaded/empty for
    ChooseBusinessView
[-] Added InputMultilineFieldCaption component and integration
[-] Moved Debouncer to separate field and refactored to the
    DispatchWorkItem and correct cancellability

Fixes #129, #130

### CODE REVIEW

File: o-fish-ios/Views/Components/Picker/SearchBarView.swift [hunk 1]
```diff
@@ -30,11 +31,10 @@ struct SearchBarView: View {
...
 
-                TextField(LocalizedStringKey(placeholder), text: $searchText, onEditingChanged: { _ in
-                    print("onEdit")
-                }, onCommit: {
-                    self.onCommit?()
-                })
+                TextField(LocalizedStringKey(placeholder),
+                    text: $searchText,
+                    onEditingChanged: onEditingChanged,
+                    onCommit: self.onCommit)
```
Comment (akaktus) [inline o-fish-ios/Views/Components/Picker/SearchBarView.swift:37]: without "self"

File: o-fish-ios/Views/ReportFlow/ChooseScreens/ChooseBusiness/ChooseBusinessView.swift [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+    /// Logic
+
+    private func loadFilteredData(searchText: String) {
+        let realmFilteredReports = RealmConnection.realm?.objects(Report.self)
+        var result = realmFilteredReports
+
+        if searchText != "" {
+            let predicate = NSPredicate(format: "vessel.lastDelivery.business CONTAINS[cd] %@ OR vessel.lastDelivery.location CONTAINS[cd] %@",
+                searchText, searchText)
```
Comment (akaktus) [inline o-fish-ios/Views/ReportFlow/ChooseScreens/ChooseBusiness/ChooseBusinessView.swift:128]: Is it really needed, using the same variable "searchText" twice?

Comment (kojiba) [reply o-fish-ios/Views/ReportFlow/ChooseScreens/ChooseBusiness/ChooseBusinessView.swift:128]: yes, it is using two %@ as arguments, 
if keep only one that last one will be (null)

File: o-fish-ios/Views/Components/Picker/SearchBarView.swift (hunk 1 above)
Comment (kojiba) [reply o-fish-ios/Views/Components/Picker/SearchBarView.swift:37]: inside the block it is need to use self

Comment (akaktus) [reply o-fish-ios/Views/Components/Picker/SearchBarView.swift:37]: But it's not a block. We have onEditingChanged closure without "self" and onCommit closure with "self"

**Pull request merged**